packages/amplify-graphql-transformer-core/src/graphql-api.ts

```diff
@@ -198,7 +205,9 @@ export class GraphQLApi extends GraphqlApiBase implements GraphQLAPIProvider {
     this.schema = props.schema ?? new TransformerSchema();
     this.schemaResource = this.schema.bind(this);
 
-    if (props.createApiKey && modes.some(mode => mode.authorizationType === AuthorizationType.API_KEY)) {
+    const hasApiKey = modes.some(mode => mode.authorizationType === AuthorizationType.API_KEY);
```

Comment:
does this check default and additional?

Reply:
Yep! https://github.com/aws-amplify/amplify-cli/blob/auth-dir-v-next/packages/amplify-graphql-transformer-core/src/graphql-api.ts#L176-L178